server/data/data-library/src/main/java/org/apache/james/user/lib/model/Algorithm.java

```diff
@@ -22,36 +22,52 @@
 import java.util.Objects;
 
 public class Algorithm {
+    private static final boolean LEGACY = true;
+    public static final String LEGACY_SUFFIX = "-legacy";
+
     public static Algorithm of(String rawValue) {
-        return new Algorithm(rawValue);
+        if (rawValue.endsWith(LEGACY_SUFFIX)) {
+            return new Algorithm(rawValue, rawValue.substring(0, rawValue.length() - LEGACY_SUFFIX.length()), LEGACY);
```

Comment:
why don't we define a closed list of support Algorithm in an enum instead?

Reply:
Because it would be a breaking change?

Because it is not related to the ISSUE addressed?

Because as a user I might want to use something stronger than SHA-512 ?

Reply:
`blabla-legacy` is not a breaking change?
I commented on some code that didn't exist before this PR, so debating the design is part of the issue AFAIC.

Reply:
Well I do not want to build an abstraction limiting the options offered by Java security implementations.

Reply:
can we at least avoid to expose the `-legacy` string and use two different factory methods?

Reply:
> can we at least avoid to expose the -legacy string

How would you do to achieve that?